src/compiler/parse/read/style.ts

```diff
@@ -16,6 +16,12 @@ export default function read_style(parser: Parser, start: number, attributes: No
 
 	const content_end = parser.index;
 
+	// discard styles when css is disabled
+	if (parser.css_mode === 'none') {
+		parser.read(/<\/style\s*>/);
```

Comment:
Maybe put the regex in a `const` so that the three (line 11 and 76) will never diverge?